lxd/storage_volumes_snapshot.go

```diff
@@ -187,13 +181,44 @@ func storagePoolVolumeSnapshotsTypePost(d *Daemon, r *http.Request) response.Res
 		return response.SmartError(err)
 	}
 
+	var parentDBVolume *db.StorageVolume
+	var parentVolumeArgs db.StorageVolumeArgs
+	err = s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+		// Get the parent volume so we can get the config.
+		parentDBVolume, err = tx.GetStoragePoolVolume(ctx, pool.ID(), projectName, volumeType, volumeName, true)
+		if err != nil {
+			return err
+		}
+
+		// We will need the parent volume config to determine the snapshot name.
+		if req.Name == "" {
+			parentVolumeArgs, err = tx.GetStoragePoolVolumeWithID(ctx, int(parentDBVolume.ID))
```

Comment:
Can you move this into the `if req.Name == "" {}` block below? This way you also don't have to export `parentVolumeArgs`.

Reply:
This would lead to the creation of a new transaction, so I think this would be slower or cause an issue... Moreover, `volumeDetermineNextSnapshotName` is already creating a transaction internally: these nested transactions will likely cause a deadlock (or in the best scenario slow down the performance). It is also possible that commands like ROLLBACK or COMMIT in the inner transaction might unexpectedly apply to the entire outer transaction.

Reply:
Ah yes you still need to access `tx.`. My mistake then :)

Reply:
Ah yes you still need to access `tx.`. My mistake then :)